funcs/func_groupcount.c

```diff
@@ -51,37 +56,200 @@
 			channel's current group if not specified (and non-empty).</para>
 		</description>
 	</function>
+	<function name="GROUP_MATCH_LIST_START" language="en_US">
+		<synopsis>Start a find of groups by regular expression</synopsis>
+		<syntax argsep="@">
+			<parameter name="group_regex"/>
+			<parameter name="category_regex"/>
+		</syntax>
+		<description>
+			<para>Once GROUP_MATCH_LIST_START is executed with a search, then each call to GROUP_MATCH_LIST_NEXT()
+			will return a matched group.  This search will look at all known groups and filter by the regex provided.</para>
+			<para>It is possible to only specify @category_regex, and it is also possible to pass in nothing for the search.</para>
+			<para>If only the category_regex is specified, then it will only return group@category that matches this category search.</para>
+			<para>If the search regex is entirely empty, all groups will be returned.</para>
+			<para>The return value is the number of groups found. Calling GROUP_MATCH_LIST_NEXT() more times than there
+			are matches will result in an empty value being returned.</para>
+			<para>This function is useful to use instead of GROUP_MATCH_LIST if you would prefer to process the results one by one.
+			If your group/category names contain a comma, then this is better to avoid parse issues</para>
+			<para>GROUP_CHANNEL_LIST does not support regex, but GROUP_MATCH_LIST_START does</para>
+			<para>Note: the search is executed and stored at the time of calling this function.
+			If GROUP() assignments change during successive calls to the _NEXT function, you will need to call _START again
```

Comment:
Instead of _NEXT and _START, use the full names.
I still don't think this function is necessary, anyone have any other thoughts?

Reply:
My thoughts are redoing this to return a hash object.

Reply:
But what does this permit users to do, that cannot already be accomplished using existing dialplan applications/functions?
I don't really think individual constructs all need their own methods of iterating over them, maybe we should improve the existing ones (separately) in that case.